tests/DoiTest.php

```diff
@@ -22,4 +32,37 @@ public function testLowercasesDois()
     {
         $this->assertEquals(['10.1097/01.asw.0000443266.17665.19'], Doi::extract('10.1097/01.ASW.0000443266.17665.19'));
     }
+
+    public function testDiscardsTrailingPunctuation()
+    {
+        $this->assertEquals(['10.1130/2013.2502'], Doi::extract('This is an example of a DOI: 10.1130/2013.2502.'));
+    }
+
+    public function testDiscardsContiguousTrailingPunctuation()
+    {
+        $this->assertEquals(['10.1130/2013.2502'], Doi::extract('This is an example of a DOI: 10.1130/2013.2502...",'));
+    }
+
+    public function testDiscardsTrailingUnicodePunctuation()
+    {
+        $this->assertEquals(['10.1130/2013.2502'], Doi::extract('This is an example of a DOI: 10.1130/2013.2502…'));
+    }
+
+    public function testRetainsClosingParenthesesThatArePartOfTheDoi()
+    {
+        $this->assertEquals(['10.1130/2013.2502(04)'], Doi::extract('This is an example of a DOI: 10.1130/2013.2502(04)'));
+    }
+
+    public function testDiscardsClosingParenthesesThatAreNotPartOfTheDoi()
+    {
+        $this->assertEquals(['10.1130/2013.2502'], Doi::extract('(This is an example of a DOI: 10.1130/2013.2502)'));
+    }
+
+    public function testExtractsExoticDois()
+    {
+        $this->assertEquals(
+            ['10.1002/(sici)1096-8644(199601)99:1<135::aid-ajpa8>3.0.co;2-#'],
+            Doi::extract('This is an example of an exotic DOI: 10.1002/(SICI)1096-8644(199601)99:1<135::AID-AJPA8>3.0.CO;2-#')
```

Comment:
Would it be worthwhile to verify that the 'exotic' DOIs can also cope with trailing punctuation?